docs-src/references/dynamic-configuration.md

```diff
@@ -0,0 +1,214 @@
+- [History Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/history/configs/config.go#L309)
+- [Matching Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/matching/config.go#L125)
+- [Worker Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/worker/service.go#L193)
+
+Note that setting dynamic configuration is optional.
+Change these values only if you need to override the default values to achieve better performance on your Temporal Cluster.
+Also, ensure that you test your changes before setting these in production.
+
+## Format
+
+To override the default dynamic configuration values, specify your custom values and constraints for the dynamic configuration keys that you want to change in a YAML configuration file.
+Use the following format when creating your dynamic configuration file.
+
+```yaml
+testGetBoolPropertyKey:
+  - value: false
+  - value: true
+    constraints:
+      namespace: "your-namespace"
+  - value: false
+    constraints:
+      namespace: "your-other-namespace"
+testGetDurationPropertyKey:
+  - value: "1m"
+    constraints:
+      namespace: "your-namespace"
+      taskQueueName: "longIdleTimeTaskqueue"
+testGetFloat64PropertyKey:
+  - value: 12.0
+    constraints:
+      namespace: "your-namespace"
+testGetMapPropertyKey:
+  - value:
+      key1: 1
+      key2: "value 2"
+      key3:
+        - false
+        - key4: true
+          key5: 2.0
+```
+
+### Constraints
+
+Some dynamic configuration keys are set globally, and others can be customized on a Namespace or Task Queue level.
+
+- To set global values for the configuration key with no constraints, use:
+
+  ```yaml
+  keyName:
+    - value: "my-value"
+  ```
+
+- For keys that can be customized at Namespace level, you can specify multiple values for different Namespaces in addition to one default value that applies globally to all Namespaces.
+  To set values at a Namespace level, use `namespace` (String) as shown in the following example.
+
+  ```yaml
+  keyName:
+    - constraints: {} # Sets default value that applies to all Namespaces
+      value: "value-for-all-the-rest"
+  - constraints: {namespace: "namespace1"} # Sets value for "namespace1" Namespace
+      value: "value-for-namespace1"
+    - constraints: {namespace: "namespace2"}
+      value: "value-for-namespace2"
+  ```
+
+- For keys that can be customized at a Task Queue level, you can specify Task Queue name and Task type in addition to Namespace.
+  To set values at a Task Queue level, use `taskqueueName` (String) with `taskType` (optional; supported values: `Workflow` and `Activity`), as shown in the following exmaple.
+
+  ```yaml
+  keyName:
+  - constraints: {namespace: "namespace1", taskQueueName: "tq"}  #Applies to both Workflow and Activity tasks on the "tq" Task Queue.
```

Comment:
I think we should say the config applies to the task queue, not to the tasks

Reply:
updated to 
` Applies to the "tq" Task Queue for both Workflows and Activities.`
and
`Applies to the "other_tq" Task Queue for Activities specifically.`